app/Nova/Actions/ExportResumes.php

```diff
@@ -0,0 +1,252 @@
+    }
+
+    /**
+     * Get the fields available on the action.
+     *
+     * @return array<\Laravel\Nova\Fields\Field>
+     */
+    public function fields(): array
+    {
+        // This is only stored for 30 seconds because it only needs to stick for one page load
+        // Nova runs this function for every record in the user view so it slows down the page otherwise
+        $majors = Cache::remember('majors_with_resumes', 30, static function (): array {
+            return User::selectRaw('distinct(majors.display_name) as distinct_display_names')
+                ->active()
+                ->whereNotNull('resume_date')
+                ->where('primary_affiliation', 'student')
+                ->whereDoesntHave('duesPackages', static function (Builder $q): void {
+                    $q->where('restricted_to_students', false);
+                })
+                ->leftJoin('major_user', static function (JoinClause $join): void {
+                    $join->on('users.id', '=', 'major_user.user_id')
+                         ->whereNull('major_user.deleted_at');
+                })
+                ->leftJoin(
+                    'majors',
+                    'major_user.major_id',
+                    '=',
+                    'majors.id'
+                )
+                ->orderBy('distinct_display_names')
+                ->pluck('distinct_display_names')
+                ->mapWithKeys(static function (?string $displayName): array {
+                    return null === $displayName ? [] : [$displayName => $displayName];
+                })
+                ->toArray();
+        });
+
+        $classStandings = Cache::remember('class_standings_with_resumes', 30, static function (): array {
+            return User::selectRaw(
+                'distinct(class_standings.name) as distinct_class_standings, class_standings.rank_order'
+            )
+                ->active()
+                ->whereNotNull('resume_date')
+                ->where('primary_affiliation', 'student')
+                ->whereDoesntHave('duesPackages', static function (Builder $q): void {
+                    $q->where('restricted_to_students', false);
+                })
+                ->leftJoin('class_standing_user', static function (JoinClause $join): void {
+                    $join->on('users.id', '=', 'class_standing_user.user_id')
+                         ->whereNull('class_standing_user.deleted_at');
+                })
+                ->leftJoin(
+                    'class_standings',
+                    'class_standing_user.class_standing_id',
+                    '=',
+                    'class_standings.id'
+                )
+                ->orderBy('class_standings.rank_order')
+                ->pluck('distinct_class_standings')
+                ->mapWithKeys(static function (?string $displayName): array {
+                    return null === $displayName ? [] : [$displayName => $displayName];
```

Comment:
Perhaps capitalize the displayed text? https://laravel.com/docs/8.x/helpers#method-fluent-str-title

Reply:
Class standings are all single words, so I just used `ucfirst`.